app.js

```diff
@@ -40,13 +30,52 @@ let allProducts = [
   new OddDuckProduct('water-can', 'img/water-can.jpg'),
   new OddDuckProduct('wine-glass', 'img/wine-glass.jpg'),
 ];
+
+// Render method
+
+OddDuckProduct.prototype.render = function(n){
+
+  let imgContainer = document.getElementById(`image${n}`);
+  imgContainer.src = (this.imgPath);
+};
+
+//Implement Local Storage / Persistent data between refreshes/restarts.
+
+
+function saveInitial(){
+
+  let savedProducts = [];
+
+  for(let i = 0; i < allProducts.length; i++) {
+    let product = allProducts[i];
+    savedProducts.push(product);
+  }
+  savedProducts = JSON.stringify(savedProducts);
+  localStorage.setItem('productData', savedProducts);
+}
+
+OddDuckProduct.prototype.loadPriorData = function (){
+
+  let savedProducts = localStorage.getItem('productData');
+
+  savedProducts = JSON.parse(savedProducts);
+  savedProducts.push(this);
+
+  for(let i = 0; i < allProducts.length; i++){
+    let product = allProducts[i];
+    savedProducts.push(product);
+  }
+  savedProducts = JSON.stringify(savedProducts);
+  localStorage.setItem('productData', savedProducts);
+
+};
```

Comment:
This method continually adds data to productData, but it never overwrites anything.